src/shell/ASTPrinter.cpp

```diff
@@ -90,6 +90,17 @@ export class ASTPrinter: public Visitor
         }
     }
 
+    void visit(VariableSubstExpr const& node) override
+    {
+        _result += "substitute";
```

Comment:
```suggestion
        _result += "varsubst";
```
maybe this, because we'll have more than one way to substitute in the future?

Reply:
Renamed to getenv